From 324f81d6f9931d2504f9eeac3f12524f70f9043d Mon Sep 17 00:00:00 2001
From: Pekka Enberg <penberg@scylladb.com>
Date: Tue, 28 Apr 2020 09:20:47 +0300
Subject: [PATCH scylla 2/4] configure.py: Add '--with-libdeflate' option

This patch adds a '--with-libdeflate=<PATH>' option to configure.py, which
allows user to override the default libdeflate submodule path. This is
useful when building packages from source tarballs, for example.
---
 configure.py               | 10 +++++++---
 sstables/checksum_utils.hh |  2 +-
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/configure.py b/configure.py
index 4edb40a020..7fb87d8f85 100755
--- a/configure.py
+++ b/configure.py
@@ -439,6 +439,7 @@ arg_parser.add_argument('--so', dest='so', action='store_true',
 arg_parser.add_argument('--mode', action='append', choices=list(modes.keys()), dest='selected_modes')
 arg_parser.add_argument('--with', dest='artifacts', action='append', choices=all_artifacts, default=[])
 arg_parser.add_argument('--with-seastar', action='store', dest='seastar_path', default='seastar', help='Path to Seastar sources')
+arg_parser.add_argument('--with-libdeflate', action='store', dest='libdeflate_path', default='libdeflate', help='Path to libdeflate sources')
 arg_parser.add_argument('--cflags', action='store', dest='user_cflags', default='',
                         help='Extra flags for the C++ compiler')
 arg_parser.add_argument('--ldflags', action='store', dest='user_ldflags', default='',
@@ -1399,7 +1400,7 @@ with open(buildfile_tmp, 'w') as f:
         f.write(textwrap.dedent('''\
             cxx_ld_flags_{mode} = {cxx_ld_flags}
             ld_flags_{mode} = $cxx_ld_flags_{mode}
-            cxxflags_{mode} = $cxx_ld_flags_{mode} {cxxflags} -I. -I $builddir/{mode}/gen
+            cxxflags_{mode} = $cxx_ld_flags_{mode} {cxxflags} -I. -I {libdeflate_path} -I $builddir/{mode}/gen
             libs_{mode} = -l{fmt_lib}
             seastar_libs_{mode} = {seastar_libs}
             rule cxx.{mode}
@@ -1445,7 +1446,10 @@ with open(buildfile_tmp, 'w') as f:
               command = $cxx -MD -MT $out -MF $out.d {seastar_cflags} $cxxflags $cxxflags_{mode} $obj_cxxflags -x c++ --include=$in -c -o $out /dev/null
               description = CHECKHH $in
               depfile = $out.d
-            ''').format(mode=mode, antlr3_exec=antlr3_exec, fmt_lib=fmt_lib, **modeval))
+            rule test.{mode}
+              command = ./test.py --mode={mode}
+              description = TEST {mode}
+            ''').format(mode=mode, antlr3_exec=antlr3_exec, fmt_lib=fmt_lib, libdeflate_path=args.libdeflate_path, **modeval))
         f.write(
             'build {mode}: phony {artifacts}\n'.format(
                 mode=mode,
@@ -1614,7 +1618,7 @@ with open(buildfile_tmp, 'w') as f:
         f.write('  pool = submodule_pool\n')
         f.write('  mode = {mode}\n'.format(**locals()))
         f.write('rule libdeflate.{mode}\n'.format(**locals()))
-        f.write('  command = make -C libdeflate BUILD_DIR=../build/{mode}/libdeflate/ CFLAGS="{libdeflate_cflags}" CC={args.cc} ../build/{mode}/libdeflate//libdeflate.a\n'.format(**locals()))
+        f.write('  command = make -C {args.libdeflate_path} BUILD_DIR=../build/{mode}/libdeflate/ CFLAGS="{libdeflate_cflags}" CC={args.cc} ../build/{mode}/libdeflate//libdeflate.a\n'.format(**locals()))
         f.write('build build/{mode}/libdeflate/libdeflate.a: libdeflate.{mode}\n'.format(**locals()))
         f.write('  pool = submodule_pool\n')
         f.write('build build/{mode}/zstd/lib/libzstd.a: ninja\n'.format(**locals()))
diff --git a/sstables/checksum_utils.hh b/sstables/checksum_utils.hh
index 925e0aa96b..3b9131fdbf 100644
--- a/sstables/checksum_utils.hh
+++ b/sstables/checksum_utils.hh
@@ -23,7 +23,7 @@
 
 #include <zlib.h>
 #include <seastar/util/gcc6-concepts.hh>
-#include "libdeflate/libdeflate.h"
+#include <libdeflate.h>
 #include "utils/gz/crc_combine.hh"
 
 GCC6_CONCEPT(
-- 
2.21.3

